pyriemann/utils/mean.py

```diff
@@ -304,6 +305,63 @@ def mean_ale(covmats, tol=10e-7, maxiter=50, sample_weight=None):
     return C
 
 
+def mean_alm(covmats, tol=1e-14, maxiter=1000,
+             verbose=False, sample_weight=None):
+    r"""Return Ando-Li-Mathias (ALM) mean
+
+    Find the geometric mean recursively [1]_, generalizing from:
+
+    .. math::
+        \mathbf{C} = A^{\frac{1}{2}}(A^{-\frac{1}{2}}B^{\frac{1}{2}}A^{-\frac{1}{2}})^{\frac{1}{2}}A^{\frac{1}{2}}
+
+    require a high number of iterations.
+
+    This is the adaptation of the Matlab code proposed by Dario Bini and
+    Bruno Iannazzo, http://bezout.dm.unipi.it/software/mmtoolbox/
+    Extremely slow, due to the recursive formulation.
+
+    :param covmats: Covariance matrices set, (n_trials, n_channels, n_channels)
+    :param tol: the tolerance to stop the gradient descent
```

Comment:
Is it really a "gradient descent" algorithm?

Reply:
Copy paste error, there is no gradient descent, this is the main point of the approach.